ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -577,6 +577,86 @@ public void deleteErrorCasesTest() {
     verifyDeleteFailure(getUniqueId(), StoreErrorCodes.ID_Not_Found);
   }
 
+  /**
+   * Test DELETE authorization success for {@link BlobStore#delete(MessageWriteSet)}.
+   * If either accountId or containerId is -1, validation is skipped.
+   */
+  @Test
+  public void deleteAuthorizationSuccessTest() throws Exception {
+    short[] accountIds = {-1, Utils.getRandomShort(TestUtils.RANDOM), -1};
+    short[] containerIds = {-1, -1, Utils.getRandomShort(TestUtils.RANDOM)};
+    for (int i = 0; i < accountIds.length; i++) {
+      MockId mockId = putSingleBlob(UtilsTest.getRandomString(10), accountIds[i], containerIds[i]);
+      MessageInfo info = new MessageInfo(mockId, DELETE_RECORD_SIZE, Utils.getRandomShort(TestUtils.RANDOM),
+          Utils.getRandomShort(TestUtils.RANDOM), System.currentTimeMillis());
+      MessageWriteSet writeSet = new MockMessageWriteSet(Collections.singletonList(info),
+          Collections.singletonList(ByteBuffer.allocate(DELETE_RECORD_SIZE)));
+      store.delete(writeSet);
+      verifyDeleteFailure(mockId, StoreErrorCodes.ID_Deleted);
+    }
+  }
+
+  /**
+   * Test DELETE authorization failure for {@link BlobStore#delete(MessageWriteSet)}.
+   */
+  @Test
+  public void deleteAuthorizationFailureTest() throws Exception {
+    if (store.getStoreConfig().storeValidateAuthorization == true) {
+      MockId mockId = put(1, PUT_RECORD_SIZE, Utils.Infinite_Time).get(0);
+      assertNotSame("accountId should not be -1 for added Blob.", -1, mockId.getAccountId());
+      assertNotSame("containerId should not be -1 for added Blob.", -1, mockId.getContainerId());
+      short[] accountIds = {-1, Utils.getRandomShort(TestUtils.RANDOM), -1, mockId.getAccountId(), Utils.getRandomShort(
+          TestUtils.RANDOM)};
+      short[] containerIds = {-1, -1, Utils.getRandomShort(TestUtils.RANDOM), Utils.getRandomShort(
+          TestUtils.RANDOM), mockId.getContainerId()};
+      for (int i = 0; i < accountIds.length; i++) {
+        verifyDeleteFailure(new MockId(mockId.getID(), accountIds[i], containerIds[i]),
+            StoreErrorCodes.Authorization_Failure);
+      }
+    }
+  }
+
+  /**
+   * Test GET authorization success for {@link BlobStore#get(List, EnumSet)}.
+   * If either accountId or containerId is -1, validation is skipped.
+   */
+  @Test
+  public void getAuthorizationSuccessTest() throws Exception {
+    short[] accountIds = {-1, Utils.getRandomShort(TestUtils.RANDOM), -1};
+    short[] containerIds = {-1, -1, Utils.getRandomShort(TestUtils.RANDOM)};
+    for (int i = 0; i < accountIds.length; i++) {
+      MockId mockId = putSingleBlob(UtilsTest.getRandomString(10), accountIds[i], containerIds[i]);
+      store.get(Collections.singletonList(mockId), EnumSet.noneOf(StoreGetOptions.class));
+      store.get(
```

Comment:
see my comment about modifying the `put()` function in this class. If you do that, you can use `checkStoreInfo` to make sure the return from these gets is correct